puzzles/solutions/2023/d01/p2.py

```diff
@@ -0,0 +1,38 @@
+import sys
+
+import p1
+
+
+DIGITS_NAMES_TO_NUMBERS = dict(
+    zip(
+        ("one", "two", "three", "four", "five", "six", "seven", "eight", "nine"),
+        (map(str, range(1, 10))),
+    )
+)
+
+
+def _convert_digits_names_to_numbers(input_text: str) -> str:
+    new_text = ""
+    for line in input_text.splitlines():
+        new_line = line[0]
+        for i in range(1, len(line)):
+            current_part_of_line = line[: i + 1]
```

Comment:
Instead of constructing new strings, you can pass `startswith`/`endswith` the start/end indices.

Reply:
Refactored in ce1f57065df2648dfabee5264d12307f934a1b67.